crypto/x509/v3_purp.c

```diff
@@ -570,14 +556,9 @@ int X509_check_ca(X509 *x)
 static int check_ssl_ca(const X509 *x)
 {
     int ca_ret;
+
     ca_ret = check_ca(x);
-    if (!ca_ret)
-        return 0;
-    /* check nsCertType if present */
-    if (ca_ret != 5 || x->ex_nscert & NS_SSL_CA)
-        return ca_ret;
-    else
-        return 0;
+    return ca_ret;
```

Comment:
This function could just be:
```
return check_ca(x);
```
But, then with this simplification, the `check_ssl_ca()` function could be replaced with `check_ca()`